packages/angular-query-experimental/src/__tests__/inject-query.test.ts

```diff
@@ -1,31 +1,255 @@
 import {
   Component,
   Injectable,
+  Injector,
   computed,
+  effect,
   inject,
   input,
   signal,
 } from '@angular/core'
 import { TestBed, fakeAsync, flush, tick } from '@angular/core/testing'
-import { QueryClient } from '@tanstack/query-core'
 import { describe, expect, vi } from 'vitest'
-import { injectQuery } from '../inject-query'
-import { provideAngularQuery } from '../providers'
+import { QueryCache, QueryClient, injectQuery, provideAngularQuery } from '..'
 import {
   delayedFetcher,
   getSimpleFetcherWithReturnData,
+  queryKey,
   rejectFetcher,
   setSignalInputs,
   simpleFetcher,
 } from './test-utils'
+import type { CreateQueryOptions, OmitKeyof, QueryFunction } from '..'
 
 describe('injectQuery', () => {
+  let queryCache: QueryCache
+  let queryClient: QueryClient
   beforeEach(() => {
+    queryCache = new QueryCache()
+    queryClient = new QueryClient({ queryCache })
     TestBed.configureTestingModule({
-      providers: [provideAngularQuery(new QueryClient())],
+      providers: [provideAngularQuery(queryClient)],
     })
   })
 
+  test('should return the correct types', () => {
```

Comment:
These are Angular's version of the correct types test for react-query:
https://github.com/TanStack/query/blob/main/packages/react-query/src/__tests__/useQuery.test.tsx